src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java

```diff
@@ -301,7 +312,50 @@ private JPanel getUninstalledAddOnsPanel() {
 		}
 		return uninstalledAddOnsPanel;
 	}
-	
+
+	private JPanel getSearchPanel(final JXTable table) {
+		JPanel searchPanel = new JPanel();
+		searchPanel.setLayout(new GridBagLayout());
+
+		JLabel lookupLabel = new JLabel(Constant.messages.getString("cfu.label.addons.filter"));
+		final JTextField lookupField = new JTextField();
+
+		searchPanel.add(lookupLabel, LayoutHelper.getGBC(0, 0, 1, 0.0D));
+		searchPanel.add(lookupField, LayoutHelper.getGBC(1, 0, 1, 1.0D));
+
+		// Set lookup listener
+		lookupField.getDocument().addDocumentListener(new DocumentListener() {
+			@Override
+			public void insertUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			@Override
+			public void removeUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			@Override
+			public void changedUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			public void updateFilter() {
+				try {
+					String lookupFieldText = lookupField.getText();
+					if (lookupFieldText.length() == 0) {
+						table.setRowFilter(null);
+					} else {
+						table.setRowFilter(RowFilter.regexFilter("(?i)" + lookupFieldText));
+					}
+				} catch (PatternSyntaxException e) {
+					logger.error("Incorrect pattern syntax " + e);

```

Comment:
It should not be logged as error, this is expected to happen, not a bug or issue in ZAP. If you really want to log the exception better to use debug level instead.

It would be great to give some indication that the regex is wrong, maybe with a visual change in the text field?